Framework/Kernel/inc/MantidKernel/StringTokenizer.h

```diff
@@ -0,0 +1,130 @@
+#ifndef MANTID_KERNEL_STRINGTOKENIZER_H_
+#define MANTID_KERNEL_STRINGTOKENIZER_H_
+
+//----------------------------------------------------------------------
+// Includes
+//----------------------------------------------------------------------
+
+#include <cstddef> //std::size_t
+#include <string>
+#include <vector>
+
+#include "MantidKernel/System.h" //DLLExport
+
+/** StringTokenizer: A simple tokenizer that splits a string into tokens, which
+ are separated by separator characters. An iterator or index can used to iterate
+ over all tokens or the result returned as a std::vector<std::string>>
+
+ Copyright © 2007-2011 ISIS Rutherford Appleton Laboratory, NScD Oak
+ Ridge National Laboratory & European Spallation Source
+
+ This file is part of Mantid.
+
+ Mantid is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 3 of the License, or
+ (at your option) any later version.
+
+ Mantid is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ File change history is stored at:
+ <https://svn.mantidproject.org/mantid/trunk/Code/Mantid>
+ Code Documentation is available at <http://doxygen.mantidproject.org>
+ */
+
+namespace Mantid {
+namespace Kernel {
+
+class DLLExport StringTokenizer {
+public:
+  /// Specify tokenizer options. These can be combined using + or |.
+  enum Options {
+    TOK_IGNORE_EMPTY = 1, ///< ignore empty tokens
+    TOK_TRIM = 2, ///< remove leading and trailing whitespace from tokens
+    TOK_IGNORE_FINAL_EMPTY_TOKEN =
+        4 ///< ignore an empty token at the end of the string.
+  };
+
+  typedef std::vector<std::string> TokenVec;
+  typedef std::vector<std::string>::iterator Iterator;
+  typedef std::vector<std::string>::const_iterator ConstIterator;
+  /// Constructs an object from an empty string.
+  StringTokenizer() = default;
+  /// Constructor requiring a string to tokenize and a string of separators.
+  StringTokenizer(const std::string &str, const std::string &separators,
+                  unsigned options = 0);
+
+  /// Destroys the tokenizer.
+  ~StringTokenizer() = default;
+
+  /** Iterator referring to first element in the container.
+  * @return an iterator referring to the first element in the container.
+  */
+  Iterator begin() { return m_tokens.begin(); };
```

Comment:
Nothing serious, but all the methods implemented in the header have an additional `;`.